packages/app/src/components/ContextSidePanel.tsx

```diff
@@ -0,0 +1,267 @@
+import { sq } from 'date-fns/locale';
+import ms from 'ms';
+import { useForm } from 'react-hook-form';
+import {
+  ChartConfigWithDateRange,
+  TSource,
+} from '@hyperdx/common-utils/dist/types';
+import { Badge, Flex, Group, SegmentedControl } from '@mantine/core';
+import { useDebouncedValue } from '@mantine/hooks';
+
+import { SQLInlineEditorControlled } from '@/components/SQLInlineEditor';
+import WhereLanguageControlled from '@/components/WhereLanguageControlled';
+import SearchInputV2 from '@/SearchInputV2';
+
+import { DBSqlRowTable } from './DBRowTable';
+
+enum ContextBy {
+  All = 'all',
+  Custom = 'custom',
+  Host = 'host',
+  Node = 'k8s.node.name',
+  Pod = 'k8s.pod.name',
+  Service = 'service',
+}
+
+interface ContextSubpanelProps {
+  source: TSource;
+  dbSqlRowTableConfig: ChartConfigWithDateRange | undefined;
+  rowData: Record<string, any>;
+  rowId: string | undefined;
+}
+
+export default function ContextSubpanel({
+  source,
+  dbSqlRowTableConfig,
+  rowData,
+  rowId,
+}: ContextSubpanelProps) {
+  const QUERY_KEY_PREFIX = 'context';
+  const { Timestamp: origTimestamp } = rowData;
+  const {
+    where: originalWhere = '',
+    whereLanguage: originalLanguage = 'lucene',
+  } = dbSqlRowTableConfig ?? {};
+  const [range, setRange] = useState<number>(ms('30s'));
+  const [contextBy, setContextBy] = useState<ContextBy>(ContextBy.All);
+  const { control, watch } = useForm({
+    defaultValues: {
+      where: '',
+      whereLanguage: originalLanguage ?? ('lucene' as 'lucene' | 'sql'),
+    },
+  });
+
+  const formWhere = watch('where');
+  const [debouncedWhere] = useDebouncedValue(formWhere, 1000);
+
+  const date = useMemo(() => new Date(origTimestamp), [origTimestamp]);
+
+  const newDateRange = useMemo(
+    (): [Date, Date] => [
+      new Date(date.getTime() - range / 2),
+      new Date(date.getTime() + range / 2),
+    ],
+    [date, range],
+  );
+
+  /* Functions to help generate WHERE clause based on
+     which Context the user chooses (All, Host, Node, etc...).
+     Since we support lucene and sql, we need to format the condition 
+     based on the language
+  */
+  const {
+    'k8s.node.name': k8sNodeName,
+    'k8s.pod.name': k8sPodName,
+    'host.name': host,
+    'service.name': service,
+  } = rowData.ResourceAttributes ?? {};
+
+  const CONTEXT_MAPPING = {
+    [ContextBy.All]: {
+      field: '',
+      value: originalWhere,
+    },
+    [ContextBy.Custom]: {
+      field: '',
+      value: debouncedWhere,
+    },
+    [ContextBy.Service]: {
+      field: 'service.name',
+      value: service,
+    },
+    [ContextBy.Host]: {
+      field: 'host.name',
+      value: host,
+    },
+    [ContextBy.Pod]: {
+      field: 'k8s.pod.name',
+      value: k8sPodName,
+    },
+    [ContextBy.Node]: {
+      field: 'k8s.node.name',
+      value: k8sNodeName,
+    },
+  } as const;
+
+  // Helper function to combine WHERE clauses
+  function combineWhereClauses(
```

Comment:
nit: i'm not sure if using `filters` would be appropriate here, but it'd be nice to just call out that `filters` is the right pattern in case anyone stumbles upon this and wants to duplicate this pattern without realizing they might benefit from `filters`

Reply:
In the middle of looking at filters right now. Will report back

Reply:
I like the idea of utilizing filters, but it's not as straightforward as it seems. I added a TODO comment in the code in case anyone comes across this code in the future.